compiler/rustc_traits/src/chalk/lowering.rs

```diff
@@ -70,114 +68,15 @@ impl<'tcx> LowerInto<'tcx, chalk_ir::AliasTy<RustInterner<'tcx>>> for ty::Projec
 }
 
 impl<'tcx> LowerInto<'tcx, chalk_ir::InEnvironment<chalk_ir::Goal<RustInterner<'tcx>>>>
-    for ChalkEnvironmentAndGoal<'tcx>
+    for ChalkGoal<'tcx>
```

Comment:
This impl should really change all that much. The `Predicate` arm becomes basically the entire body of the `filter_map` for clauses.